flask_influxdb/flask_influxdb.py

```diff
@@ -0,0 +1,183 @@
+import influxdb
+from flask import current_app, _app_ctx_stack, Flask
+from flask.globals import _app_ctx_err_msg
+from typing import Generator
+
+_no_influx_msg = """\
+No Influx connection is present.
+
+This means that something has overwritten _app_ctx_stack.top.influxdb_db.
+"""
+
+
+class InfluxDB(object):
+    def __init__(self, app: Flask = None) -> None:
+        """
+        Class constructor
+        :param app: Flask Application object
+        """
+        self.app = app
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app: Flask) -> None:
+        """
+        Initialize extension for application
+        :param app: Flask Application object
+        :return:
+        """
+        app.config.setdefault('INFLUXDB_HOST', 'localhost')
+        app.config.setdefault('INFLUXDB_PORT', '8086')
+        app.config.setdefault('INFLUXDB_USER', 'root')
+        app.config.setdefault('INFLUXDB_PASSWORD', 'root')
+        app.config.setdefault('INFLUXDB_DATABASE', None)
+        app.config.setdefault('INFLUXDB_SSL', False)
+        app.config.setdefault('INFLUXDB_VERIFY_SSL', False)
+        app.config.setdefault('INFLUXDB_RETRIES', 3)
+        app.config.setdefault('INFLUXDB_TIMEOUT', None)
+        app.config.setdefault('INFLUXDB_USE_UDP', False)
+        app.config.setdefault('INFLUXDB_UDP_PORT', 4444)
+        app.config.setdefault('INFLUXDB_PROXIES', None)
+        app.config.setdefault('INFLUXDB_POOL_SIZE', 10)
+
+        app.teardown_appcontext(self.teardown)
+
+        with app.app_context():
+            self.database.create(app.config['INFLUXDB_DATABASE'])
```

Comment:
This crashes if `INFLUXDB_DATABASE` is not set (because the default is None). It might be worth making it conditional based on if `INFLUXDB_DATABASE` is set.